src/InstrumentationEngine.Lib/InstrumentationMethodAttachContext.cpp

```diff
@@ -0,0 +1,26 @@
+#include "stdafx.h"
+#include "InstrumentationMethodAttachContext.h"
+
+namespace MicrosoftInstrumentationEngine
+{
+    CInstrumentationMethodAttachContext::CInstrumentationMethodAttachContext(_In_ IEnumInstrumentationMethodSettings* pSettingsEnum) noexcept :
+        m_pSettingsEnum(pSettingsEnum)
+    {
+    }
+
+    STDMETHODIMP CInstrumentationMethodAttachContext::QueryInterface(_In_ REFIID riid, _Out_ void** ppvObject)
```

Comment:
>(_In_ REFIID riid, _Out_ void** ppvObject) [](start = 68, length = 42)

do you need to add "noexcept" to these as well, to avoid warnings?

Reply:
I don't think so. I don't see it on IUnknown::QueryInterface and I don't see build warnings for this in the PR build nor on my local machine. What warning do you expect?

---
In reply to: [382173711](https://github.com/microsoft/CLRInstrumentationEngine/pull/204#discussion_r382173711) [](ancestors = 382173711)

Reply:
Sometimes you get warnings about declarations mismatching definitions with this kind of thing. If you don't get a warning, that's fine.

---
In reply to: [382213914](https://github.com/microsoft/CLRInstrumentationEngine/pull/204#discussion_r382213914) [](ancestors = 382213914,382173711)